### PR TITLE
Medications

### PR DESCRIPTION
**Pull request by jonathandick**
Creating first commit for Medications Widget. 

Presently, neither the fhir nor restws apis are providing for the strength of the medication. 

Primarily patterned this off the Vitals Widget.

image

### CODE REVIEW

File: src/summary/history/medications/medications-level-one.component.tsx [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+        <tbody>
+          {patientMedications &&
+            patientMedications.map((medication, index) => {
+              return (
+                <React.Fragment key={medication.id}>
+                  {medication.status === "active" && (
+                    <tr>
+                      <td>
+                        <span className="omrs-bold">
+                          {medication.medicationReference.display}
```
Comment (fatmali) [inline src/summary/history/medications/medications-level-one.component.tsx]: wonder why you're putting this empty space here, html is whitespace insensitive

File: src/summary/history/medications/medications-level-one.component.tsx [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+                        </span>{" "}
+                        - {medication.dosageInstruction[0].route.text} -{" "}
```
Comment (fatmali) [inline src/summary/history/medications/medications-level-one.component.tsx]: for hyphens, we are using unicode to represent them `{\u2014}`

File: src/summary/history/medications/medications-level-one.component.tsx [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+                        </svg>
+                      </td>
+                    </tr>
+                  )}
+                </React.Fragment>
+              );
+            })}
+        </tbody>
+      </table>
+      <SummaryCardFooter linkTo={`/patient/${patientUuid}/chart/Medications`} />
```
Comment (fatmali) [inline src/summary/history/medications/medications-level-one.component.tsx]: nit: I think the `M` in `/Medications` should be small caps.

File: src/summary/history/medications/medications-level-one.component.tsx (hunk 2 above)
Comment (jonathandick) [reply src/summary/history/medications/medications-level-one.component.tsx]: Updated.

File: src/summary/history/medications/medications-level-one.component.tsx (hunk 3 above)
Comment (jonathandick) [reply src/summary/history/medications/medications-level-one.component.tsx]: updated

**Comment by jonathandick** [comment]
A few notes

1. Neither the Fhir nor restws api's currently has the strength of the drug which is a critical element of the medication statement. For now, I'm leaving that out and will add in once the backend is updated (Bett is working on updating the restws api).

2. I'm using restws api. The fhir module has been identified as having too many problems and a fhir2 module will be created. Until we have fhir2, we should just be using the restws api.

3. I've renamed the cards to be more descriptive of their function. What was level 1 is now called overview; level 2 is not called "primary" as it's the primary card for showing data related to that item. 

4. For the primary card, I just have a placeholder right now uses the overview card. I will update that in a subsequent PR.

File: src/summary/documentation/vitals-card.component.tsx [hunk 4]
```diff
@@ -7,6 +7,7 @@ import { formatDate } from "./dimension-helpers";
 import { createErrorHandler } from "@openmrs/esm-error-handling";
 import { useCurrentPatient } from "@openmrs/esm-api";
 import SummaryCardFooter from "../cards/summary-card-footer.component";
+import { link } from "fs";
```
Comment (fatmali) [inline src/summary/documentation/vitals-card.component.tsx]:
```suggestion

```

File: src/summary/history/medications/medications-primary.component.tsx [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+    const subscription = fetchPatientMedications(patientUuid).subscribe(
+      Medications => setPatientMedications(Medications),
+      createErrorHandler()
+    );
+
+    return () => subscription.unsubscribe();
+  }, [patientUuid]);
+
+  return (
+    <div className={styles.medicationsLevelTwo}>
```
Comment (fatmali) [inline src/summary/history/medications/medications-primary.component.tsx]: why still use the `medicationsLevelTwo` not `medicationsPrimary`?

File: src/summary/documentation/vitals-card.component.tsx (hunk 4 above)
Comment (jonathandick) [reply src/summary/documentation/vitals-card.component.tsx]: Done

File: src/summary/history/medications/medications-primary.component.tsx (hunk 5 above)
Comment (jonathandick) [reply src/summary/history/medications/medications-primary.component.tsx]: Updated.

**Pull request merged**